CalPatRec/src/CalHelixFinderAlg.cc

```diff
@@ -2092,8 +2094,19 @@ namespace mu2e {
       printInfo(Helix);
     }
 
+    circleHits = Helix._sxy.qn();
+
     rc = doLinearFitPhiZ(Helix, HitInfo_t(StrawId::_ntotalfaces-1,0,-1), useIntelligentWeight);
 
+    phiHits = Helix._szphi.qn();
+
+    nHitsRatio = circleHits/(phiHits + 1e-6);
+
+    if (_diag > 0) {
+      Helix._diag.nHitsRatio = nHitsRatio;
+    }
+
+    if (nHitsRatio > _maxNHitsRatio) goto PATTERN_RECOGNITION_END;
     //2017-10-05 Gianipez added the following line to make some tests
     if (Helix._szphi.qn() == 0.)                                 goto  PATTERN_RECOGNITION_END;
```

Comment:
goto is a deprecated programming construct. Please replace these with nest if blocks.

Reply:
I think that this comment is arguable and C++ supports the `goto` command. I understand that some people might think that it is not simple to read, but its use is not wrong and people should be more open to different code styles.

Reply:
I agree with Dave that these not appropriate uses of goto.  Modern flow control constructs exist so that the compiler can better understand the intent of the programmer and produce better optimized code and better check for errors.

Reply:
I believe this is a legit use of `goto` as it makes crystal clear w/o the need for any comments where the condition control ends. I debugged this module many times with Valgrind and I never faced any issue with these `goto` commands. I understand that they can be replaced with nested ifs, but IMHO in this way the code is clearer. So unless there are proofs of inefficiencies or bugs, these comments are speculations driven by a style choice.

Reply:
As discussed outside this thread we will grandfather the use of goto in this class for this PR. The downsides of goto programming style are real but should be discussed in a separate forum.